### DISCUSSION TITLE
How add preview domain in vercel.

### DISCUSSION THREAD

**Discussion by MeetBhingradiya** [vercel/community#2312] [Help] [answered]
### Summary

+ I went to test beta version but vercel has own encrypeted domain that not working in my app as cors errors and every push creates a new domain.

**Comment by amyegan** [comment, marked as answer]
Hi @MeetBhingradiya! You can assign a custom domain to preview deployments from a specific branch, like in our guide about [how to set up a staging environment on Vercel](https://vercel.com/guides/set-up-a-staging-environment-on-vercel). There are other options, but that seems like the most appropriate for your needs.

There are also predictable patterns for [Git branch deployment URLs](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) and [CLI deployment URLs](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-vercel-cli). But these would still have some variation, so the CORS errors would continue. It's also possible to add a [preview deployment suffix](https://vercel.com/docs/concepts/deployments/generated-urls#preview-deployment-suffix) if you prefer that approach.

**Reply by MeetBhingradiya** [reply]
I have Already Dynamic CORS System that access Limited amount of origins after deployment i can easily modify it. that was not in my case. 

When your answer is come i do more search and i got this in Vercel > Project > Deployments

Per Deployment has a right 3 doted menu i click on this
I got option assign a domain that solved my problem. 

Thanks for helping

Best Regards,
Meet Bhingradiya
